README.md

```diff
@@ -1,45 +1,148 @@
+## Architecture
 
 Refer to the [specification](docs/SPEC.md) document.
 
+## Repo Structure
+
+```
+.
+├── README.md                       # This file
+├── main.go                         # TestGround entrypoint file
+├── cmd                             # TestGround CLI comamnds
+│   ├── all.go
+│   ├── build.go
+│   ├── daemon.go
+│   ├── enum_flag.go
+│   ├── helpers.go
+│   ├── list.go
+│   ├── run.go
+│   └── sidecar.go
+├── manifests                       # Manifests for each test Plan. These exist independent from plans to enable plans to live elsewhere
+│   ├── dht.toml
+│   └── smlbench.toml
+├── plans                           # The Test Plan. Includes Image to be run, Assertions and more
+│   ├── dht
+│   └── smlbench
+├── sdk                             # SDK available to each test plan
+│   ├── README.md
+│   ├── iptb
+│   ├── runtime
+│   └── sync
+├── docs                            # Documentation of the project
+│   ├── SPEC.md
+│   ├── container-hierarchy.png
+│   ├── dashboard-mockup.png
+│   ├── system-architecture.png
+│   └── testcase-runtime-env.png
+├── pkg                             # Internals to TestGround
+│   ├── api
+│   ├── auto
+│   ├── build
+│   ├── engine
+│   ├── logging
+│   ├── runner
+│   ├── state
+│   └── util
+└── tools                           # ??
+    └── src_generate.go
+```
+
+## Contributing & Testing
+
+We kindly ask you to read through the SPEC first and give this project a run first in your local machine. It is a fast moving project at the moment and it might require some tinkering and experimentation to compesate the lack of documentation.
+
+### Setup
+
+Ensure that you are running go 1.13 or later (for gomod support)
+
+```sh
+> go version
+go version go1.13.1 darwin/amd64
+```
+
+Download the repo and install the dependencies
+
+```sh
+> go get git@github.com:ipfs/testground.git
+# ..fetch and install logs
+> cd $GOPATH/src/github.com/ipfs/testground
+```
+
+Test that everything is installed correctly by running
+
+```sh
+> TESTGROUND_BASEDIR=`pwd` testground
+resolved testground base dir from env variable: /Users/imp/code/go-projects/src/github.com/ipfs/testground
+NAME:
+   testground - A new cli application
+
+   USAGE:
+      testground [global options] command [command options] [arguments...]
+
+   COMMANDS:
+      run      (builds and) runs test case with name `testplan/testcase`
+      list     list all test plans and test cases
+      build    builds a test plan
+      help, h  Shows a list of commands or help for one command
+
+   GLOBAL OPTIONS:
+      -v          verbose output (equivalent to INFO log level)
+      --vv        super verbose output (equivalent to DEBUG log level)
+     --help, -h  show help
+```
+
+### Running the tests locally
+
+To run a test locally, you can use the `testground run` command. Check what Test Plans are available in the `plans` folder
+
+```
+> ls plans
+dht      smlbench
+```
+
+Then do
+
+```
+> TESTGROUND_BASEDIR=`pwd` testground run plans/dht
```

Comment:
This is not working for me at the moment. Trying:

<img width="866" alt="image" src="https://user-images.githubusercontent.com/1211152/65854471-6e1b5d80-e365-11e9-97d9-69d38daa845f.png">


```

Reply:
Error comes from https://github.com/ipfs/testground/blob/master/pkg/engine/engine.go#L131-L134

Reply:
Chasing the tail of this one but must confess that one letter variables and lack of docs does make it hard. It all might just be my lack of go proficiency though. Getting there :)

Reply:
The comment says it shouldn't be instantiated directly (which I read, outside) -- https://github.com/ipfs/testground/blob/master/pkg/engine/census.go#L11-L23 -- but isn't what is being done by engine https://github.com/ipfs/testground/blob/master/pkg/engine/engine.go#L54 ?

Reply:
It seems that the map that maps Test Plans to names is found here https://github.com/ipfs/testground/blob/master/pkg/engine/census.go#L16 and it gets populated at https://github.com/ipfs/testground/blob/master/pkg/engine/census.go#L34-L36 when a TestPlan is enrolled

Reply:
However, the name is indeed `dht` at the moment https://github.com/ipfs/testground/blob/master/manifests/dht.toml#L1 and so `TESTGROUND_BASEDIR=`pwd` testground run dht` should do the trick.

Reply:
One question to answer is "are the test plans enrolled" done by https://github.com/ipfs/testground/blob/master/pkg/engine/engine.go#L78-L80.

Reply:
AH! It's not the path, it is `<testplan>/<testcase>` which is exactly what the help text said (well, it says testplan/testcase without the angle brackets) ! Feel really dumb now ahahah, I totally assumed that the first part was the folder and the second the name of the plan.